subprojects/common/retrace/src/main/kotlin/com/avito/retrace/ProguardRetracer.kt

```diff
@@ -0,0 +1,36 @@
+package com.avito.retrace
+
+import proguard.retrace.ReTrace
+import java.io.File
+import java.io.LineNumberReader
+import java.io.PrintWriter
+import java.io.StringReader
+import java.io.StringWriter
+
+interface ProguardRetracer {
+
+    fun retrace(content: String): String
+
+    class Impl(
+        private val mappings: List<File>
+    ) : ProguardRetracer {
+
+        override fun retrace(content: String): String {
+            var result = content
```

Comment:
It's strange we write into `result` in `forEach` but then return one value
Should we somehow filter mappings to find firstone

Reply:
We have multiple mapping. In logs can be several stacktraces that mapped into different mapping.txt. So this code iterate over all available mappings and trying to apply them to log.

Reply:
I understand that behavior. But It looks like we rewriting previous results in forEach and we will see only the last write to `result` variable

Reply:
We sequentially apply mappings to logcat. I don't know another away except iterating over mappings and apply them to logcat one-by-one. And updating variable with logcat on every iteration.

Could you please provide piece of code that looks good to you?

Reply:
`retraces.first { }` or `retraces.scan { }`

Reply:
Why `scan` is better than `forEach`?

Reply:
I consider that Retrace API doesn't help us.

Reply:
Furthermore `scan` is an experimental API. Is this any significant reason to use it?